src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5461,6 +5522,17 @@ public override IOperation VisitMethodReference(IMethodReferenceOperation operat
 
         public override IOperation VisitPropertyReference(IPropertyReferenceOperation operation, int? captureIdForResult)
         {
+            // Check if this is an anonymous type property reference with an implicit receiver within an anonymous object initializer.
+            // Due to https://github.com/dotnet/roslyn/issues/22736,
+            // Instance is null for property reference with implicit instance reference within an anonymous initializer.
+            if (operation.Instance == null &&
+                operation.Property.ContainingType.IsAnonymousType &&
+                operation.Property.ContainingType == _currentImplicitInstance.AnonymousType &&
+                _currentImplicitInstance.AnonymousTypePropertyCaptureIds.TryGetValue(operation.Property, out int captureId))
```

Comment:
>_currentImplicitInstance.AnonymousTypePropertyCaptureIds.TryGetValue(operation.Property, out int captureId) [](start = 16, length = 107)

Should we create an invalid operation when we don't yet have a value for this reference instead of falling back to the default rewrite?